src/microbit/model/display.py

```diff
@@ -13,8 +13,11 @@ class Display:
     def __init__(self):
         self.__image = Image()
         self.__on = True
-        self.__blank_image = Image()
         self.__current_pid = None
+        self.__blank_image = Image()
+
+        self.__image_lock = threading.Lock()
+        self.__thread_identifier_lock = threading.Lock()
 
     def scroll(self, value, delay=150, wait=True, loop=False, monospace=False):

```

Comment:
Do we know how `set_pixel` interacts with `scroll/show(wait=False)`?

Reply:
will add lock to every part that accesses self.__image

Reply:
Just curious, what would happen if you called `set_pixel` while it's scrolling?

Reply:
Looks like it sets the pixel for a brief moment, but the scrolling keeps happening and the pixel is updated to whatever it should be while it is scrolling

Reply:
My implementation works the same way :D